rainbow/devices/stm32.py

```diff
@@ -16,18 +16,52 @@
 #
 # Copyright 2019 Victor Servant, Ledger SAS
 
+import random
+
 import unicorn as uc
-import capstone as cs
-import pickle
 
-from binascii import hexlify
+from ..generics import rainbow_cortexm
+
+
+class rainbow_stm32(rainbow_cortexm):
+    """STM32 generic device
+
+    STMicroelectronics STM32 shares most peripherals addresses accross the family.
+    """
+
+    RNG_BASE_ADDR = 0x50060800
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Map RNG peripheral (Random Number Generator)
+        self.map_space(self.RNG_BASE_ADDR, self.RNG_BASE_ADDR + 0x12)
+        self.emu.hook_add(
+            uc.UC_HOOK_MEM_READ,
+            self.rng_sr_read,
+            begin=self.RNG_BASE_ADDR + 0x4,
+            end=self.RNG_BASE_ADDR + 0x4,
+        )
+        self.emu.hook_add(
+            uc.UC_HOOK_MEM_READ,
+            self.rng_dr_read,
+            begin=self.RNG_BASE_ADDR + 0x8,
+            end=self.RNG_BASE_ADDR + 0x8,
+        )
+
+    def rng_sr_read(self, _uci, _access, address, _size, _value, _user_data):
+        """Hook called before RNG status register is read"""
+        self[address] = 0x1  # data ready
 
-from rainbow.generics import rainbow_cortexm
-from rainbow.color_functions import color
+    def rng_dr_read(self, _uci, _access, address, _size, _value, _user_data):
+        """Hook called before RNG data register is read
 
+        Please feel free to override me to implement custom random values.
+        """
+        self[address] = random.randint(0, 2**32)
```

Comment:
`randint` does include upper bound, this should be `2**32 - 1`.

Reply:
Thank you, I got misled by numpy.random.randint which does not seem to include upper bound :sweat_smile: